File: src/main/java/co/com/fsighttech/imp/issuing/bo/core/Coverage.java
/**
 * 
 */
package co.com.fsighttech.imp.issuing.bo.core;

import java.util.List;

import co.com.fsighttech.imp.issuing.bo.configuration.MasterCoverage;
import lombok.Getter;
import lombok.Setter;

/**
 * @author jtorres
 *
 */
@Getter
@Setter
public class Coverage {


	private Long id; 
	private MasterCoverage masterCoverage;
	private String concept; // A code that represents the coverage concept
	private String limit; // Coverage's Limit
	private Deductible deductible; // Coverage's Deductible
	private Double totalPremium; // Coverage's total premium
	private List<VariableData> additionalFields;
	private String state;
	private Boolean contracted =false;
	private Boolean required = false;
}

File: src/main/resources/static/src/app/issuing-module/shared/model/core/coverage.ts
import { VariableData } from './variable-data';
import { MasterCoverage } from '../configuration/master-coverage';
/**
* this object represents a policy coverage
*
* @author jtorres
*/
export class Coverage {

    id: number;
    masterCoverage: MasterCoverage;
    concept: string; // A code that represents the coverage concept
    limit: string; // Coverage's Limit
    totalPremium: number; // Coverage's total premium
    additionalFields: VariableData[];
    auditFlag: string;
    state: string;
    contracted: boolean;
    required:boolean;
}